### PR TITLE
[Feature][Ops][Misc] Support batch-shared sampling and remove reduce sample

### PR DESCRIPTION
**Pull request by zouzy5137**
### What this PR does / why we need it?
As mentioned in #15119 and previous release note, this pull request removes the experimental `enable_reduce_sample` (reduce sampling) feature and all its associated code, configurations, Triton kernels, tests, and documentation. This cleanup simplifies the codebase by removing unused or deprecated experimental paths for tensor parallel communication reduction during sampling.

### Does this PR introduce _any_ user-facing change?
Yes, the experimental configuration option `enable_reduce_sample` is removed, and it can no longer be enabled in `additional_config`.
Enable batch shareded with `--enable_batch_sharded_sampling` 

### How was this patch tested?
The patch was tested by removing the corresponding unit tests and ensuring that the remaining test suite passes.
Tested with Qwen3.5-27B
#### Accuracy
with batch-shared
| dataset | version | metric | mode | vllm-api-general-chat |
|----- | ----- | ----- | ----- | -----|
| gsm8k | 50861a | accuracy | gen | 91.36 |

without batch-shared
| dataset | version | metric | mode | vllm-api-general-chat |
|----- | ----- | ----- | ----- | -----|
| gsm8k | 50861a | accuracy | gen | 90.14 |

##### Concurrency 4
with batch-shared
Common Metric | Stage | Value
-- | -- | --
Benchmark Duration | total | 165134.4871 ms
Total Requests | total | 32
Failed Requests | total | 0
Success Requests | total | 32
Concurrency | total | 3.9143
Max Concurrency | total | 4
Request Throughput | total | 0.1938 req/s
Total Input Tokens | total | 112320
Prefill Token Throughput | total | 5901.8483 token/s
Total Generated Tokens | total | 48000
Input Token Throughput | total | 680.1729 token/s
Output Token Throughput | total | 290.6722 token/s
Total Token Throughput | total | 970.8451 token/s

without batch-shared
Common Metric | Stage | Value
-- | -- | --
Benchmark Duration | total | 152077.9668 ms
Total Requests | total | 32
Failed Requests | total | 0
Success Requests | total | 32
Concurrency | total | 3.9457
Max Concurrency | total | 4
Request Throughput | total | 0.2104 req/s
Total Input Tokens | total | 112320
Prefill Token Throughput | total | 7967.7283 token/s
Total Generated Tokens | total | 48000
Input Token Throughput | total | 738.5685 token/s
Output Token Throughput | total | 315.6276 token/s
Total Token Throughput | total | 1054.1961 token/s

##### Concurrency 16
with batch-shared
Common Metric | Stage | Value
-- | -- | --
Benchmark Duration | total | 52620.9764 ms
Total Requests | total | 32
Failed Requests | total | 0
Success Requests | total | 32
Concurrency | total | 15.2525
Max Concurrency | total | 16
Request Throughput | total | 0.6081 req/s
Total Input Tokens | total | 112320
Prefill Token Throughput | total | 2593.1853 token/s
Total Generated Tokens | total | 48000
Input Token Throughput | total | 2134.51 token/s
Output Token Throughput | total | 912.1838 token/s
Total Token Throughput | total | 3046.6938 token/s

without batch-shared
Common Metric | Stage | Value
-- | -- | --
Benchmark Duration | total | 54178.6397 ms
Total Requests | total | 32
Failed Requests | total | 0
Success Requests | total | 32
Concurrency | total | 15.3356
Max Concurrency | total | 16
Request Throughput | total | 0.5906 req/s
Total Input Tokens | total | 112320
Prefill Token Throughput | total | 2634.1206 token/s
Total Generated Tokens | total | 48000
Input Token Throughput | total | 2073.1417 token/s
Output Token Throughput | total | 885.958 token/s
Total Token Throughput | total | 2959.0998 token/s

- vLLM main: https://github.com/vllm-project/vllm/commit/84030bbe3d74d99bad477a3d2e37a973ccd8865c

### CODE REVIEW

File: vllm_ascend/models/deepseek_v4/model.py [hunk 1]
```diff
@@ -1075,6 +1075,12 @@ def compute_logits(
         logits = self.logits_processor(self.lm_head, hidden_states)
         return logits
 
+    def compute_logits_local(
```
Comment (freyfwt) [inline vllm_ascend/models/deepseek_v4/model.py:1105]: model_vl also needs this function

Comment (zouzy5137) [reply vllm_ascend/models/deepseek_v4/model.py:1105]: Ok, I will fix this.

**Comment by zouzy5137** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E

**Comment by zouzy5137** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E

**Comment by zouzy5137** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E

**Comment by zouzy5137** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E

**Comment by zouzy5137** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E

**Review by linfeng-yuan — COMMENTED** [review]
Please supplement your performance and accuracy variation.

**Comment by zouzy5137** [comment]
> Please supplement your performance and accuracy variation.

OK. I will submit my test results later.

**Comment by zouzy5137** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E

**Comment by zouzy5137** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E

**Comment by linfeng-yuan** [comment]
/nightly MiniMax-M3-W8A8-A3
nightly command triggered.
- Nightly-A3: [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/35306391142)

**Comment by zouzy5137** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- Docs link check

**Failed:**
- E2E: still in progress, retry /rerun after completion

**Comment by zouzy5137** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- Docs link check

**Failed:**
- E2E: still in progress, retry /rerun after completion

**Comment by shiqiangA** [comment]
/nightly MiniMax-M3-W8A8-A3
nightly command triggered.
- Nightly-A3: [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/35333488667)

**Comment by zouzy5137** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E

**Comment by shiqiangA** [comment]
/nightly MiniMax-M3-W8A8-A3
nightly command triggered.
- Nightly-A3: [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/35480831883)

**Comment by shiqiangA** [comment]
/nightly MiniMax-M3-W8A8-A3
nightly command triggered.

**Comment by shiqiangA** [comment]
/nightly MiniMax-M3-W8A8-A3
nightly command triggered.

**Comment by shiqiangA** [comment]
/nightly MiniMax-M3-W8A8-A3 --a3-560t
nightly command triggered.
- Nightly-A3-560T: [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/35674867517)

**Comment by linfeng-yuan** [comment]
/nightly MiniMax-M3-W8A8-A3 --a3-560t
nightly command triggered.
- Nightly-A3-560T: [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/35695432891)
